primitives/core/src/crypto.rs

```diff
@@ -918,6 +925,13 @@ impl<'a> TryFrom<&'a str> for KeyTypeId {
 	}
 }
 
+/// An identifier for a specific cryptographic algorithm used by a key pair
+#[derive(Copy, Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct CryptoTypeId(pub &'static str);
```

Comment:
Didn't we say, this should better be a bound sized `u8` so we can do `Encode/Decode` without Problems?
```suggestion
pub struct CryptoTypeId(pub [u8; 4]);
```